mlflow/data/arrow_dataset.py

```diff
@@ -0,0 +1,71 @@
+from typing import Optional, Union
+
+import pyarrow as pa
+
+from mlflow.data.dataframe_dataset import DataFrameDataset, from_dataframe
+from mlflow.data.dataset_source import DatasetSource
+
+
+class ArrowDataset(DataFrameDataset[pa.Table]):
+    """Represents a PyArrow Table for use with MLflow Tracking."""
+
```

Comment:
This is simply a container, the check for `df` to be of the correct class happens in `from_<backend>`. If datasets can be used directly, then we might want to perform validation here as well